src/mcp_agent/server/app_server.py

```diff
@@ -1426,20 +1445,27 @@ async def _workflow_run(
 
 
 async def _workflow_status(
-    ctx: MCPContext, run_id: str, workflow_name: str | None = None
+    ctx: MCPContext, run_id: str | None = None, workflow_id: str | None = None
 ) -> Dict[str, Any]:
     # Ensure upstream session so status-related logs are forwarded
     try:
         _set_upstream_from_request_ctx_if_available(ctx)
     except Exception:
         pass
+
+    if not (run_id or workflow_id):
+        raise ValueError("Either run_id or workflow_id must be provided.")
+
     workflow_registry: WorkflowRegistry | None = _resolve_workflow_registry(ctx)
 
     if not workflow_registry:
         raise ToolError("Workflow registry not found for MCPApp Server.")
 
-    workflow = await workflow_registry.get_workflow(run_id)
-    workflow_id = workflow.id if workflow and workflow.id else workflow_name
+    if not workflow_id:
+        workflow = await workflow_registry.get_workflow(
+            run_id=run_id, workflow_id=workflow_id
+        )
+        workflow_id = workflow.id if workflow else None
```

Comment:
This is also changing the behavior. Should it still default to workflow_name? 
```suggestion
        workflow_id = workflow.id if workflow and workflow.id else workflow_name
```